### ISSUE TITLE
Build institutional stats page (when plugin is active)

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks-multi-institution#67]
As an institutional manager, I want to see a version of the network stats page with information that's specifically relevant to usage at my institution.

Expected behaviour:
- [x] Add a H1 to the page for super admins/network managers that reads 'Network Stats'
- [x] Add a H1 to the page for institutional manager that reads '[Institution Name] Stats'
- [x] All CSVs generated by the page view & download data `Download CSV` option should include a column called Institution and display the book's institutional assignment
- [x] Clicking the Download CSV button in Page view & download data as an Institutional Manager should result in a CSV that only includes books assigned to my institution
- [x] The Users Over Time, User Revisions Over Last, Books Over Time, Network Storage, and Most Cloned Books visualizations and Download CSV options for Institutional Managers should only include data for books and users assigned to their institution.
- [x] Institutional Managers should only be able to view their Institution's Stats page. They should not be able to view the Network Stats page or the Insitutitional Stats pages for any institutions which they do not manage

## Notes
When prioritizing which visualizations are most important, start at top and work down the page. Koko Downloads is most important, users & books over time are next, then user revisions over time, then cloned books, and network storage last. It's least important for institutional managers.

**Comment by richard015ar** [comment]
WIP:
- https://github.com/pressbooks/pressbooks-multi-institution/compare/feat/institutional-manager-stats?expand=1
- https://github.com/pressbooks/pressbooks-network-analytics/compare/feat/add-stats-hooks?expand=1

So far I added hooks for the page title (Network/Institution Name Stats), and for filtering the CSV. I'll have to add the institution name field as a next step. And much more to come next week.

**Comment by richard015ar** [comment]
WIP still in: 

- https://github.com/pressbooks/pressbooks-multi-institution/compare/feat/institutional-manager-stats?expand=1
- https://github.com/pressbooks/pressbooks-network-analytics/compare/feat/add-stats-hooks?expand=1

Done:
- Network Title
- Download pageview data filtered for IMs with institution column
- Users and Book Over Time filtered with relevant data for IMs.
- Filenames filtered `<Institution name> Books/Users Over Time` instead of `Network...`
- Most Revisions Over Time and Most Cloned Books filtered for IMs. Download actions are also filtered.
- Tests in this plugin covering all the above.

Pending:
- Change the name of the CSV filename for Most Revisions Over Time and Most Cloned Books.
- Add Institution column for mentioned stats.
- Network Storage filtered.
- Download CSV (with filename change, filtered data and Institution column added).

Progress can be tested on https://ricardodev.pressbooks.network/

**Comment by tw77** [comment]
I did some testing on ricardodev of the WIP. Looking good, just a couple of things I noticed:

I cloned a book from my institution as an IM within the network in order to test updating the Books Over Time stats. Total Books increased by 1, but Cloned Books didn't increase.

<img width="825" alt="Screenshot 2024-03-11 at 10 47 44 PM" src="https://github.com/pressbooks/pressbooks-multi-institution/assets/44182629/299576bf-6422-47d3-8ee5-0c15f215da09">

<img width="822" alt="Screenshot 2024-03-11 at 10 55 14 PM" src="https://github.com/pressbooks/pressbooks-multi-institution/assets/44182629/ecbaa988-20e4-43d9-9001-66afc9426e62">

This might be a pre-existing issue, but it seems like the y-axis for Books Over Time should only use integers since there can never be 0.5 of a book for example.

**Comment by richard015ar** [comment]
Thank you for the early testing, @tw77 . I'll investigate and see if it's a preexisting bug. I think it might, but I'll double-check.

**Comment by SteelWagstaff** [comment]
The y-axis issue with decimal display is preexisting behaviour and outside the scope of this ticket.

**Comment by tw77** [comment]
The only thing I'm not sure about is the cause of the separate issue I saw: Total Books number incrementing but the Cloned Books number not incrementing.

**Comment by richard015ar** [comment]
It is weird because it works on my local. I think it might be we need to run the analytics cron to count on new clones. I will take a look.

For my record, I think I will need to restructure the hooks a little bit. I noticed some abstractions I can do on the code. I'll work on it later today and tomorrow.

**Comment by SteelWagstaff** [comment]
@tw77 if you log in as super admin and run `https://ricardodev.pressbooks.network/wp-admin/network/admin.php?page=pb_network_analytics_booklist_sync` and then reload the stats page, is the count corrected?

**Comment by tw77** [comment]
This had no effect on the count. However, I'm about to try cloning a book from a different network to ricardodev to see what happens there. The issue I saw was when cloning a book from the network within the network.

**Comment by tw77** [comment]
Cloning a book from outside ricardodev to ricardodev also didn't change the Cloned Books count in Books Over Time that I'm seeing as an IM.

To summarize, as an IM viewing Stats I haven't been able to get the Cloned Books count to increase. All other counts in Books Over Time (Total Books, Public Books, Private Books) are able to increase as expected.

**Comment by tw77** [comment]
Oh, the Cloned Books count just updated! Though only by 1. And it took a bit longer to update than the other counts. 

Because the Cloned Books count is still 1 less than expected, I suspect the clone of a book from an outside network _to_ ricardodev was counted, but the clone of a book from inside ricardodev was not. 

I will check to see if that occurs as a network manager on Staging, for reference.

**Comment by tw77** [comment]
I've found that this is a pre-existing issue outside the scope of this ticket. Cloning a book from a network within the same network doesn't increment the Cloned Books count in the Stats page for that network. Only clones from other networks to the network are included in that count, currently. This occurs as a network manager on https://staging.pressbooks.network/ just as I encountered it as an IM on ricardodev.

**Comment by richard015ar** [comment]
Thank you @tw77 for testing and investigating the most cloned books issue. I think I have an idea about the reason for it, but I am not sure.

PRs ready for review in:
- https://github.com/pressbooks/pressbooks-network-analytics/pull/343
- https://github.com/pressbooks/pressbooks-multi-institution/pull/89
Both dependant.

@tw77 I addressed all the pending items I mentioned earlier. These changes are updated now on ricardodev and ready for testing in case you want to do it before it is merged. I left testing notes at https://github.com/pressbooks/pressbooks-multi-institution/pull/89#issue-2182668699. Just note that I refreshed the DB on my dev network to get the latest DB changes (Dalcin's PR)
